### PR TITLE
Enable RSA and ECC with multiple signing alg

### PR DESCRIPTION
**Pull request by jeskosda**
This PR should try out to easily add EC as supported alg for token validation without big refactoring of existing code.

A breaking change would be to not accept empty alg in the jwks, because this needs to be distinguished now. 

I also need to add BouncyCastle if i want to easily resolve the Alg from the PEM file, at least i found no better solution. 
I we don't want to have the dependency, then we could still allow only RS265 for the PEM files. Are PEM files used somewhere? ALH? SI? Wondering if that is still a desired feature?

### CODE REVIEW

File: sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/service/AuthService.java [hunk 1]
```diff
@@ -1,17 +1,117 @@
...
+    }
+  }
+
+  private Optional<DecodedJWT> verifyJwtSignature(
+      String authorizationToken, LoadedPublicKey loadedPublicKey) {
+    try {
+
+      Verification jwtVerification =
+          JWT.require(resolveAlgorithm(loadedPublicKey)).acceptLeeway(leeway);
+
```
Comment (jeskosda) [inline sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/service/AuthService.java:101]: This is just the new part.

File: sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/JwksKeySource.java [hunk 2]
```diff
@@ -156,6 +204,9 @@ private static class Key {
     private String use;
     private String n;
     private String e;
+    private String x;
+    private String y;
+    private String crv;
```
Comment (christopher-cudennec) [inline sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/JwksKeySource.java:209]: Are these common/good names for EC?

**Review by christopher-cudennec — COMMENTED** [review]
Does your PR still contain a breaking change? Could you point me to the class/line of code?
Have you tried the PR in another application?

File: sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/JwksKeySource.java (hunk 2 above)
Comment (jeskosda) [reply sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/JwksKeySource.java:209]: https://datatracker.ietf.org/doc/html/rfc7517#section-3 that's how it's provided

File: sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/JwksKeySource.java [hunk 3]
```diff
@@ -97,39 +111,73 @@ private boolean isForSigning(Key key) {
...
+    return SUPPORTED_KTY.contains(key.getKty());
+  }
+
+  private boolean isSupportedAlg(Key key) {
+    return SUPPORTED_ALG.contains(key.getAlg());
   }
 
-  private boolean isRsa256Key(Key key) {
-    // We only support RSA256, if blank we assume it to be RS256.
-    return StringUtils.isBlank(key.getAlg()) || "RS256".equals(key.getAlg());
```
Comment (jeskosda) [inline sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/JwksKeySource.java:106]: The breaking change is actually here. So far we assume if the `"alg": "" ` is provided in the JWK, the RSA256 is used as default. Which is in my opinion a bad behavior. The "alg" should be "none" (also not supported so far) or have an existing alg. 

We would throw an exception if none of the supported algs couldnt be determined.

Comment (jeskosda) [reply sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/JwksKeySource.java:106]: We could still fallback to `RSA256`, but this feels not right in my opinion. The parameter is OPTIONAL in the specification: https://datatracker.ietf.org/doc/html/rfc7517#section-4.4, which means for me that we are using a not signed JWT so not a JWS here. Which is the equivalent of "none". Which could be an proper default value then. 

See the list of algs here: https://www.iana.org/assignments/jose/jose.xhtml

**Comment by jeskosda** [comment]
Regarding test: It could be a bit more difficult here. First of all we need to provide JWK with EC and provide a token which is signed with ES256. But i will investigate if our keycloak can do that 👍🏻

File: sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/JwksKeySource.java (hunk 3 above)
Comment (JoergSiebahn) [reply sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/JwksKeySource.java:106]: For security reasons we should not support `alg: "none"`.

Comment (christopher-cudennec) [reply sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/key/JwksKeySource.java:106]: We do fix a non-compliant implementation by this "breaking change", don't we? We could argue that this is not a breaking change then. What do you think?

**Comment by jeskosda** [comment]
@christopher-cudennec  I tested the SNAPSHOT against our keycloak with a token using signing alg ES256. Works fine 👍🏻

**Review by christopher-cudennec — APPROVED** [review]

**Pull request merged**